### ISSUE TITLE
using the controller for ppm and fanspeed

### ISSUE THREAD

**Issue by checkiecheck** [soloam/ha-pid-controller#40]
So I see the output in documentation is in same unit as sensor. But , i want to control the fanspeed of a ventilation unit to keep the Co2 ppm on a specific level.
Should min and max output then be in PPM or in RPM (or % of max speed which is easier)

so sensor.co2 = ppm
output.min= 5 (% of max rpm)
output.max = 95 (% of max rpm)

OR
output.min = 400ppm
output.max=1400ppm

but then i dont know what to put the fan speed at. 
Could you please enlighten me before i plunge into a wrong direction?

**Comment by Cor1984** [comment]
I would say that an output of 0-100% is best. In most cases the controller is used to control something (like fan speed) sticking to just a range in percentage gives the user the freedom to control whatever device he wants to control.

**Comment by soloam** [comment]
Mostly I use the output from 0-100% as @Cor1984 stated, then I calculate the   percentage accordingly. But if you prefer, you could set the minimum to 400 and the max to 1400, that that would be the rpm to pass to the fan.

**Comment by checkiecheck** [comment]
yeah.. thx both.. Did some more reading on PID controllers in general. I did go with the 100% approach. Ventilator has hard time (capacity wise) to pull ppm levels down when more people are present, so i have no good way to adjust the parameters.. will play for a few days from here (though setpoint changes are disrupting the controller i have the feeling)
